sdk/keyvault/keyvault-admin/test/internal/serviceVersionParameter.spec.ts

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as assert from "assert";
+import { createSandbox, SinonSandbox, SinonSpy } from "sinon";
+import { KeyVaultAccessControlClient, KeyVaultBackupClient } from "../../src";
+import { LATEST_API_VERSION } from "../../src/constants";
+import { HttpClient, WebResourceLike, HttpOperationResponse, HttpHeaders } from "@azure/core-http";
+import { ClientSecretCredential } from "@azure/identity";
+import { env } from "@azure/test-utils-recorder";
+
+// Adding this to the source would change the public API.
+type ApIVersions = "7.2-preview";
+
+describe("The keyvault-admin clients should set the serviceVersion", () => {
+  const keyVaultUrl = `https://eastus2.keyvault_name.managedhsm-int.azure-int.net`;
+
+  function makeHTTPMock(path: string, status = 200): HttpClient {
+    return {
+      async sendRequest(httpRequest: WebResourceLike): Promise<HttpOperationResponse> {
+        return {
+          status,
+          headers: new HttpHeaders(),
+          request: httpRequest,
+          parsedBody: {
+            id: `${keyVaultUrl}${path}`,
+            attributes: {}
+          }
+        };
+      }
+    };
+  }
+
+  let mockHttpClient: HttpClient;
+  let sandbox: SinonSandbox;
+  let spy: SinonSpy<[WebResourceLike], Promise<HttpOperationResponse>>;
+  let credential: ClientSecretCredential;
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  describe("KeyVaultAccessControlClient", () => {
+    beforeEach(async () => {
+      mockHttpClient = makeHTTPMock("/providers/Microsoft.Authorization/roleDefinitions");
+      sandbox = createSandbox();
+      spy = sandbox.spy(mockHttpClient, "sendRequest");
+
+      credential = await new ClientSecretCredential(
+        env.AZURE_TENANT_ID!,
+        env.AZURE_CLIENT_ID!,
+        env.AZURE_CLIENT_SECRET!
+      );
+    });
+
+    it("it should default to the latest API version", async function() {
+      const client = new KeyVaultAccessControlClient(keyVaultUrl, credential, {
+        httpClient: mockHttpClient
+      });
+      await client.listRoleDefinitions("/").next();
+      const calls = spy.getCalls();
+      assert.equal(
+        calls[0].args[0].url,
+        `${keyVaultUrl}///providers/Microsoft.Authorization/roleDefinitions?api-version=${LATEST_API_VERSION}`
+      );
+    });
+
+    it("it should allow us to specify an API version from a specific set of versions", async function() {
+      const serviceVersion = "7.2-preview";
+      const credential = await new ClientSecretCredential(
```

Comment:
`credential` is already created in `beforeEach`. is it needed to be created again?

Reply:
Good point! Thank you!

Reply:
I'll remove this.